app/controllers/residential_address_controller.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+class ResidentialAddressController < StandardStepsController
+  def edit
+    @step = step_class.new(snap_application_attributes.slice(*step_attrs))
+  end
```

Comment:
you are right! I initially had this looking pretty different from `StandardStepsController` but then realized I could just overwrite `snap_application_attributes` and it would work. 

Removed at  ba02d66